src/pages/components/CertificateInfo.qml

```diff
@@ -12,63 +12,61 @@ import Sailfish.Silica.private 1.0 as Private
 import Sailfish.Browser 1.0
 import Qt5Mozilla 1.0
 import "." as Browser
+import Sailfish.WebView.Controls 1.0
 
 SilicaFlickable {
     id: root
     property QMozSecurity security
     readonly property bool _validCert: security && security.subjectDisplayName.length > 0
-    contentHeight: certInfoColumn.height + certInfoColumn.y + showMore.height + calculatedHeight + 2 * Theme.paddingSmall
+    contentHeight: certInfoColumn.height
     clip: contentHeight > height
-    property alias buttonHeight: showMore.height
     readonly property bool _secure: security && security.allGood
-    property bool portrait
-    property bool initialized
-    property int implicitKeyWidth: Theme.itemSizeExtraLarge
-    property int keyWidth: implicitKeyWidth
-    property int maximumKeyWidth: width / 2
-    property int calculatedHeight
 
     signal showCertDetail
     signal closeCertInfo
 
-    function updateKeyWidth() {
-        if (!initialized) return
-
-        var width = implicitKeyWidth
-        for (var i in keyValuePairs.children) {
-            var child = keyValuePairs.children[i]
-            if (child.keyLabel !== undefined) {
-                var childWidth = Math.ceil(child.keyLabel.implicitWidth)
-                if (childWidth > maximumKeyWidth) continue
-                width = Math.max(width, childWidth)
-            }
-        }
-        keyWidth = width
-
-        // Column won't layout until the next frame, so we have to calculate the height ourselves
-        calculatedHeight = 0
-        for (var i in keyValuePairs.children) {
-            var child = keyValuePairs.children[i]
-            if ((child.keyLabel !== undefined) && (child.value.length > 0)) {
-                calculatedHeight += child.height + Theme.paddingSmall
+    function updatePermissionIndicationModel() {
+        permissionIndicationModel.clear()
+
+        PermissionManager.requestUriPermissions(toolBarRow.url, function(permissions) {
+            for (var i in permissions) {
+                var type = permissions[i].type
+                var capability = parseInt(permissions[i].capability)
+                if (capability === PermissionManager.Prompt || capability === PermissionManager.Unknown)
+                    continue
+
+                var source = "image://theme/icon-m-location"
```

Comment:
You should colorize this like "image://theme/icon-m-location?#FF0000" <= red being and example. Icons must not be colorized by default rather colorizing shall be done on the software.